crates/agglayer-node/src/kernel/mod.rs

```diff
@@ -329,12 +329,23 @@ where
         &self,
         signed_tx: &SignedTx,
     ) -> Result<TransactionReceipt, SettlementError<RpcProvider>> {
-        let hash = signed_tx.hash().to_string();
+        let hex_hash = signed_tx.hash();
+        let hash = format!("{:?}", hex_hash);
+
         let f = self
             .build_verify_batches_trusted_aggregator_call(signed_tx)
             .await
             .map_err(SettlementError::ContractError)?;
 
+        if let Ok(Some(tx)) = self.check_tx_status(hex_hash).await {
+            warn!(hash, "Transaction already settled: {:?}", tx);
+        }
+
+        info!(
```

Comment:
maybe debug?

Reply:
For now I let it as `info` in order to have it on the logs in datadog, but after the debugging phase we can reduce its level.